docs/en/ingest-management/fleet/add-fleet-server-kubernetes.asciidoc

```diff
@@ -0,0 +1,392 @@
+
+[discrete]
+[[add-fleet-server-kubernetes-cert-prereq]]
+== {fleet-server} and SSL/TLS certificates considerations
+
+This section shows the minimum requirements in terms of Transport Layer Security (TLS) certificates for the {fleet-server}, assuming no mutual TLS (mTLS) is needed. Refer to <<tls-overview>> and <<mutual-tls>> for more information about the configuration needs of both approaches.
+
+There are two main and differentiated traffic flows associated with the {fleet-server}, where each of them has different TLS needs:
+
+[discrete]
+[[add-fleet-server-kubernetes-cert-inbound]]
+=== {agent} → {fleet-server} inbound traffic flow
+
+In this flow the {fleet-server} acts as the server and the {agent} acts as the client, therefore the {fleet-server} requires a TLS certificate and key, and the {agent} will need to trust the CA certificate used to sign the {fleet-server} certificate.
+
+[NOTE]
+====
+A {fleet-server} certificate is not required when installing the server using the *Quick start* mode, but should always be used for *production* deployments. In *Quick start* mode, the {fleet-server} uses a self-signed certificate and the {agent}s have to be enrolled with the `--insecure` option.
+====
+
+If your organization already uses the {stack}, you may already have a CA certificate that could be used to generate the new cert for the {fleet-server}. If you do not have a CA certificate, you can read more about generating a CA and a server certificate for the {fleet-server} in <<generate-fleet-server-certs>>.
+
+You also need to know and plan in advance the <<fleet-server-hosts-setting, hostname / URL>> that the {agent} clients will use to access the {fleet-server}, because the *hostname* part of the URL needs to be included in the server certificate as an `x.509 Subject Alternative Name (SAN)`. If you plan to make your {fleet-server} accessible through *multiple hostnames* or *FQDNs*, add all of them to the server certificate.
+
+[discrete]
+[[add-fleet-server-kubernetes-cert-outbound]]
+=== {fleet-server} → {es} output outbound traffic flow
+
+In this flow the {fleet-server} acts as the client and {es} acts as the HTTPS server. For the communication to succeed, the {fleet-server} needs to trust the CA certificate used to sign the {es} certificate. If your {es} uses certificates signed by a corporate CA or multiple intermediate CAs you will need to use them during the {fleet-server} setup.
+
+[NOTE]
+====
+If your {es} cluster is on Elastic Cloud or if it uses a certificate signed by a public and known CA you won't need the {es} CA during the setup.
+====
+
+In summary, you need:
+
+* A *server certificate and key*, valid for the {fleet-server} URL. The CA used to sign this certificate will be needed by the {agent} clients, not by the {fleet-server} itself.
+
+* The *CA certificate* (or certificates) associated to your {es} cluster, except if you are sure your {es} certificate is fully trusted publicly.
```

Comment:
@bmorelli25 : do you think that `summary` is a bit redundant? In theory it's a summary of both traffic flows needs, and in the introduction we also mention both items (cert+key and CA cert of ES).

Reply:
Hmmm. Yes and no. I can see a case for keeping this and for removing it. I don't feel strongly either way.

Reply:
Ok! we will keep this for the moment and review at a later stage ;)